.github/workflows/cpp-packaging.yml

```diff
@@ -668,88 +637,44 @@ jobs:
-      uses: actions/download-artifact@v2.0.8
-      with:
-        name: firebase_cpp_sdk.zip
-    - name: Unzip Firebase C++ SDK
-      shell: bash
-      run: |
-        ls
-        if [ "$RUNNER_OS" == "Windows" ]; then
-          7z x firebase_cpp_sdk.zip
-        else
-          unzip -q firebase_cpp_sdk.zip
-        fi
+
     - name: Setup python
       uses: actions/setup-python@v2
       with:
         python-version: '3.7'
-    - name: Add msbuild to PATH (windows)
-      if: runner.os == 'Windows'
-      uses: microsoft/setup-msbuild@v1.0.2
-    - name: Prepare for integration tests
-      run: |
-        pip install -r scripts/gha/requirements.txt
-        python scripts/gha/restore_secrets.py --passphrase "${{ secrets.TEST_SECRET }}"
-    - name: Build integration tests
+    - name: Use expanded matrix
+      if: github.event_name == 'schedule' || github.event.inputs.use_expanded_matrix == '1'
       run: |
-        python scripts/gha/build_testapps.py --t ${{ env.apis }} --p ${{ matrix.target_platform }} --packaged_sdk firebase_cpp_sdk --output_directory "${{ github.workspace }}" --noadd_timestamp
-
-    - name: Run desktop integration tests
-      if: matrix.target_platform == 'Desktop' && !cancelled()
-      run: |
-        python scripts/gha/desktop_tester.py --testapp_dir testapps
-    # Workaround for https://github.com/GoogleCloudPlatform/github-actions/issues/100
-    # Must be run after the Python setup action
-    - name: Set CLOUDSDK_PYTHON (Windows)
-      shell: bash
-      if: runner.os == 'Windows' && !cancelled()
-      run: echo "CLOUDSDK_PYTHON=${{env.pythonLocation}}\python.exe" >> $GITHUB_ENV
-    - name: Install Cloud SDK
-      if: ${{ !cancelled() }}
-      uses: google-github-actions/setup-gcloud@master
-    - name: Upload Desktop Artifacts to GCS
-      if: ${{ !cancelled() }}
-      run: |
-        python scripts/gha/gcs_uploader.py --testapp_dir testapps --key_file scripts/gha-encrypted/gcs_key_file.json
-    - name: Run mobile integration tests
-      if: matrix.target_platform != 'Desktop' && !cancelled()
-      run: |
-        python scripts/gha/test_lab.py --android_model ${{ env.android_device }} --android_api ${{ env.android_api }} --ios_model ${{ env.ios_device }} --ios_version ${{ env.ios_version }} --testapp_dir testapps --code_platform cpp --key_file scripts/gha-encrypted/gcs_key_file.json
-    - name: Summarize build and test results
-      if: ${{ !cancelled() }}
+        echo "USE_EXPANDED_MATRIX=1" >> $GITHUB_ENV
+    - name: Generate token for GitHub API
+      # This step is necessary because the existing GITHUB_TOKEN cannot be used inside one workflow to trigger another.
+      # 
+      # Instead, generate a new token here, using our GitHub App's private key and App ID (saved as Secrets).
+      # 
+      # This method is preferred over the "personal access token" solution, as the GitHub App's scope is limited to just
+      # the firebase-cpp-sdk repository.
```

Comment:
I'm super duper glad this works. It reduces so much maintenance.

Reply:
It's a neat workaround! I'm glad we don't have to use anyone's personal token.